pyvrp/cpp/DurationSegment.h

```diff
@@ -132,25 +137,48 @@ DurationSegment DurationSegment::merge(Matrix<Duration> const &durationMatrix,
     using Dur = pyvrp::Duration;
 
     // edgeDuration is the travel duration from our last to the other's first
-    // client, and atOther the time (after starting from our first client) at
-    // which we arrive there.
+    // client
     Dur const edgeDuration = durationMatrix(idxLast_, other.idxFirst_);
-    Dur const atOther = duration_ - timeWarp_ + edgeDuration;
 
-    // Time warp increases if we arrive at the other's first client after its
-    // time window closes, whereas wait duration increases if we arrive there
-    // before opening.
-    Dur const diffTw = std::max<Dur>(twEarly_ + atOther - other.twLate_, 0);
-    Dur const diffWait = other.twEarly_ - atOther > twLate_
-                             ? other.twEarly_ - atOther - twLate_
-                             : 0;  // ternary rather than max avoids underflow
+    // We must wait if we arrive before the earliest start time of the other
+    // segment. Note that the earliest start time of the other segment may not
+    // be the opening of the time window of the first client, but rather the
+    // time we should start to avoid unnecessary waiting time. We may still
+    // start earlier at that other segment, but this suggests that we should
+    // wait at another moment for the same total duration since we cannot
+    // finish the other route segment earlier.
+    Dur const waitDuration
```

Comment:
This is still a diff/delta in wait duration, right?

Reply:
Yes, in a sense. It is the duration you have to wait between the two segments that are merged, just like `edgeDuration` is the travel duration between the segments. You could consider `waitDuration` a delta for the total wait duration, but similarly `edgeDuration` is a delta for the total edge duration (or travel duration). Hence I think it makes sense to simply call it `waitDuration` in the context of the merge.